File: Lighthouse/Models/Throughput.cs
namespace Lighthouse.Models
{
    public class Throughput
    {
        public Throughput() : this([])
        {
        }

        public Throughput(int[] throughputPerUnitOfTime) 
        {
            ThroughputPerUnitOfTime = throughputPerUnitOfTime;
        }

        public int[] ThroughputPerUnitOfTime { get; set; }

        public int History => ThroughputPerUnitOfTime.Length;

        public int GetThroughputOnDay(int day)
        {
            return ThroughputPerUnitOfTime[day];
        }
    }
}

File: Lighthouse.Frontend/src/models/Forecasts/Throughput.ts
export interface IThroughput {
	history: number;
	throughputPerUnitOfTime: number[];
	totalThroughput: number;
	getThroughputOnDay(day: number): number;
}

export class Throughput implements IThroughput {
	throughputPerUnitOfTime: number[];
	history: number;
	totalThroughput: number;

	constructor(
		throughputPerUnitOfTime: number[],
		history: number,
		totalThroughput: number,
	) {
		this.throughputPerUnitOfTime = throughputPerUnitOfTime;
		this.history = history;
		this.totalThroughput = totalThroughput;
	}

	getThroughputOnDay(day: number): number {
		if (day < 0 || day >= this.throughputPerUnitOfTime.length) {
			throw new RangeError("Invalid day index.");
		}

		return this.throughputPerUnitOfTime[day];
	}
}
